compiler/qsc/src/interpret/stateful.rs

```diff
@@ -414,6 +422,90 @@ impl Interpreter {
         Ok(result)
     }
 
+    /// # Errors
+    /// If the parsing of the line fails, an error is returned.
+    /// If the compilation of the line fails, an error is returned.
+    /// If there is a runtime error when generating code for the line, an error is returned.
+    pub fn qirgen_line(&mut self, line: &str) -> Result<String, Vec<LineError>> {
+        if self.target != TargetProfile::Base {
+            return Err(vec![LineError(WithSource::new(
+                line.into(),
+                LineErrorKind::TargetMismatch,
+                None,
+            ))]);
+        }
+
+        let mut codegen = BaseProfGen::new();
+
+        let mut fragments = self.compiler.compile_fragments(line).map_err(|errors| {
```

Comment:
I think we should specifically use the expr parser and reject anything that's not an expression here. IMO it makes for a really weird user experience if we allow the user to pass in statements / namespace declarations inside `qsharp.compile(xxx)`. The signature at the Python layer currently defines this as an "entry expression" which makes sense to me from a user's perspective.

If the user wants to declare additional operations or whatever, they can do that outside of `compile()` , in a regular `%%qsharp` cell or `qhsarp.eval()` call.

I'm also motivated by the fact that we plan to add `qsharp.eval_with_shots(xxx)`, and that will also be restricted to expressions (to keep things sane). It would make a coherent api if these two functions had similar calling patterns.

Reply:
That sounds like a good, cohesive story. It should be an easy change to make in the code that will simplify things quite a bit, too. Let me try that in the other branch, and we can see if that one is clean enough to warrant a PR to supersede this one.